sources/packages/api/src/route-controllers/application/application.students.controller.ts

```diff
@@ -448,4 +456,120 @@ export class ApplicationStudentsController extends BaseController {
       applicationNumber: application.applicationNumber,
     };
   }
+
+  /**
+   * Get in progress details of an application by application id.
+   * @param applicationId application id.
+   * @returns application details.
+   */
+  @Get(":id/in-progress")
+  @ApiNotFoundResponse({
+    description: "Application id not found.",
+  })
+  async getInProgressApplicationDetails(
+    @Param("id", ParseIntPipe) applicationId: number,
+  ): Promise<InProgressApplicationDetailsAPIOutDTO> {
+    const application = await this.applicationService.getApplicationByIdAndUser(
+      applicationId,
+    );
+    if (!application) {
+      throw new NotFoundException(
+        `Application id ${applicationId} was not found.`,
+      );
+    }
+
+    const incomeVerificationDetails =
+      await this.craIncomeVerificationService.allIncomeVerificationsForAnApplication(
+        applicationId,
+      );
+    const incomeVerification =
+      this.applicationControllerService.processApplicationIncomeVerificationDetails(
+        incomeVerificationDetails,
+      );
+
+    const supportingUserDetails =
+      await this.supportingUserService.getSupportingUsersByApplicationId(
+        applicationId,
+      );
+
+    const supportingUser =
+      this.applicationControllerService.processApplicationSupportingUserDetails(
+        supportingUserDetails,
+      );
+
+    return {
+      id: application.id,
+      applicationStatus: application.applicationStatus,
+      pirStatus: application.pirStatus,
+      PIRDeniedReason: getPIRDeniedReason(application),
+      offeringStatus: application.currentAssessment?.offering.offeringStatus,
+      exceptionStatus: application.applicationException?.exceptionStatus,
+      ...incomeVerification,
+      ...supportingUser,
+    };
+  }
+
+  /**
+   * Get cancelled details of an application by application id.
+   * @param applicationId application id.
+   * @returns application details.
+   */
+  @Get(":id/cancelled")
+  @ApiNotFoundResponse({
+    description: "Application id not found.",
+  })
+  async getCancelledApplicationDetails(
+    @Param("id", ParseIntPipe) applicationId: number,
+  ): Promise<CancelledApplicationDetailsAPIOutDTO> {
+    const application = await this.applicationService.getApplicationByIdAndUser(
```

Comment:
We must check if the student has access to the application and we should not rely on the userId since the same student can have multiple ones.

Reply:
Please referer to comment https://github.com/bcgov/SIMS/pull/1414#discussion_r982751253